src/domain/task-system/services/src/task_scheduler_impl.rs

```diff
@@ -52,12 +50,18 @@ impl TaskSchedulerImpl {
 impl TaskScheduler for TaskSchedulerImpl {
     #[tracing::instrument(level = "info", skip_all, fields(?logical_plan))]
     async fn create_task(&self, logical_plan: LogicalPlan) -> Result<TaskState, CreateTaskError> {
-        let mut task = Task::new(
-            self.time_source.now(),
-            self.event_store.new_task_id().await?,
-            logical_plan,
-        );
-        task.save(self.event_store.as_ref()).await.int_err()?;
+        let task = <DatabaseTransactionRunner>::run_transactional_with(
```

Comment:
See my concerns above, I don't think `TaskScheduler` is a right kind of component to control transactions.